lib/sensu-plugins-postgres/pgpass.rb

```diff
@@ -0,0 +1,17 @@
+module Pgpass
+  def pgpass
+    if File.file?(config[:pgpass])
+      pgpass = Hash[[:hostname, :port, :database, :user, :password].zip(File.read(config[:pgpass]).strip.split(':'))]
+      pgpass[:database] = nil if pgpass[:database] == '*'
```

Comment:
When database in pgpass is defined as `*` which means all databases I set it to nil in order to expect database name either from `-d` argument or `PGDATABASE` variable. It should not break iteration. You would get `config['database'] = nil` and `PG.connect()` would throw exception.

Defaults are not removed. They are moved to the `pgpass()` function. It is needed because now we are going to have 4 sources of data:  command line pgpass file env variables and defaults.

Importance levels are as follows starting from the most important:
1. command line arguments
2. pgpass file
3. env variables
4. defaults

If you set defaults during argument parsing you cannot distinguish user inputted value from default value. This is the reason I removed default values from `option`.

When I use pgpass file I expect that hostname and port are taken from the file but I also want to have possibility to override it with CLI arguments.

I hope that explains your concerns.